### ISSUE TITLE
`FBIvarReference *wrapper = [[FBIvarReference alloc] initWithIvar:ivar];`  crashed when ivar is NULL

### ISSUE THREAD

**Issue by suxinde2009** [facebook/FBRetainCycleDetector#72]
hi, dudes 
      
I am hitting a situation that when detecting an instance retain cycle references,  the 
**FBIvarReference *wrapper = [[FBIvarReference alloc] initWithIvar:ivar];**  in function 
**NSArray<id<FBObjectReference>> *FBGetClassReferences(Class aCls)**
was called, and the **ivar** was NULL by coincidence, then the program crashes because of EXC_BAD_ACCESS, so I added

```
if (typeEncoding == NULL) {
    return FBUnknownType;
  }

```

in front of 

```
if (typeEncoding[0] == '{') {
    return FBStructType;
  }

```
in method **- (FBType)_convertEncodingToType:(const char *)typeEncoding** of **FBIvarReference**  class. My problem was resolved then, so I am just curious if this defensive code should be added.

Any useful reply will be appreciated. ^_^

**Comment by rs17b** [comment]
#76

**Comment by rs17b** [comment]
Fixed in #76
